# Исследование надежности заемщиков


**Описание проекта**

Заказчик — кредитный отдел банка. Нужно разобраться, влияет ли семейное положение и количество детей клиента на факт погашения 
кредита в срок. Входные данные от банка — статистика о платёжеспособности клиентов.
Результаты исследования будут учтены при построении модели кредитного скоринга — специальной системы, которая оценивает
способность потенциального заёмщика вернуть кредит банку.

**Описание данных**

- **children**— количество детей в семье
- **days_employed** — общий трудовой стаж в днях
- **dob_years** — возраст клиента в годах
- **education** — уровень образования клиента
- **education_id** — идентификатор уровня образования
- **family_status** — семейное положение
- **family_status_id** — идентификатор семейного положения
- **gender** — пол клиента
- **income_type** — тип занятости
- **debt** — имел ли задолженность по возврату кредитов
- **total_income** — ежемесячный доход
- **purpose** — цель получения кредита

**План работ**

1. Изучить общую информацию о данных
2. Выполнить предобработку данных:
    - найти и обработать пропуски в данных,
    - найти и обработать аномальные значения,
    - найти дубликаты и удалить их,
    - проверить типы данных
3. Категоризировать данные
4. Исследовать данные и зависимости между ними
5. Оформить результаты, сделать общий вывод


    

Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [111]:
#загрузка библиотек Pandas
import pandas as pd

In [112]:
try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [113]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [114]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [115]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [116]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [117]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [118]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [119]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [120]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [121]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [122]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [123]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [124]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру.**

In [125]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [126]:
data.duplicated().sum()

71

In [127]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [128]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [129]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [130]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [131]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [132]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [133]:
dolg = data.groupby('children')['debt'].sum() #считаем число должников по каждой группе в зависимости от числа детей
kol = data.groupby('children')['gender'].count()#считаем общее кол-во клиентов в каждой группе в зависимости от числа детей
print(dolg)
print(kol)
for i in data['children'].unique():
    convers = (dolg/kol)*100   #считаем конверсию 
convers
   



children
0    1063
1     444
2     194
3      27
4       4
5       0
Name: debt, dtype: int64
children
0    14091
1     4808
2     2052
3      330
4       41
5        9
Name: gender, dtype: int64


children
0    7.543822
1    9.234609
2    9.454191
3    8.181818
4    9.756098
5    0.000000
dtype: float64

In [134]:
data_grouped = data.groupby(['children']).agg({'gender':'count','debt':'sum'})
data_grouped['dolg'] = data_grouped['debt'] / data_grouped['gender'] *100
data_grouped

gender  debt      dolg
children                        
0          14091  1063  7.543822
1           4808   444  9.234609
2           2052   194  9.454191
3            330    27  8.181818
4             41     4  9.756098
5              9     0  0.000000

In [135]:
data_pivot = data.pivot_table(index=['children'], values=('debt'), aggfunc=('count','sum'))
data_pivot['dolg']=data_pivot['sum']/data_pivot['count']*100
data_pivot

count   sum      dolg
children                       
0         14091  1063  7.543822
1          4808   444  9.234609
2          2052   194  9.454191
3           330    27  8.181818
4            41     4  9.756098
5             9     0  0.000000

**Вывод:** Посчитаем конверсию, т.е. отношение числа должников с определенным количеством детей к общему числу клиентов с таким же количеством детей. Из полученных результатов можно сделать вывод, что больше всего задолженностей у тех клиентов, кто имеет 1,2 или 4 детей. По полученным результатм видно, что нет задолженностей среди клиентов, у которых 5 детей, но здесь статистика по данной группе является не очень показательной, так как слишком мала,поэтому данный вывод не является корректным. Исходя из всего можно сделать вывод, что наименьшее кол-во задолженностей среди клиентов без детей.   

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [136]:
dol_s = data.groupby('family_status')['debt'].sum() #считаем кол-во должников в зависимости от семейного положения
kol_s = data.groupby('family_status')['gender'].count() #считаем общее кол-во клиентов в зависимости от СП
print(dol_s)
print(kol_s)
for i in data['family_status'].unique():
    convers_s = (dol_s/kol_s)*100   #считаем конверсию 
convers_s

family_status
Не женат / не замужем    273
в разводе                 84
вдовец / вдова            63
гражданский брак         385
женат / замужем          927
Name: debt, dtype: int64
family_status
Не женат / не замужем     2796
в разводе                 1189
вдовец / вдова             951
гражданский брак          4134
женат / замужем          12261
Name: gender, dtype: int64


family_status
Не женат / не замужем    9.763948
в разводе                7.064760
вдовец / вдова           6.624606
гражданский брак         9.313014
женат / замужем          7.560558
dtype: float64

In [137]:
data_grouped_s = data.groupby(['family_status']).agg({'gender':'count','debt':'sum'})
data_grouped_s['dolg_s'] = data_grouped_s['debt'] / data_grouped_s['gender'] *100
data_grouped_s

gender  debt    dolg_s
family_status                                
Не женат / не замужем    2796   273  9.763948
в разводе                1189    84  7.064760
вдовец / вдова            951    63  6.624606
гражданский брак         4134   385  9.313014
женат / замужем         12261   927  7.560558

In [138]:
data_pivot_s = data.pivot_table(index=['family_status'], values=('debt'), aggfunc=('count','sum'))
data_pivot_s['dolg_s']=data_pivot_s['sum']/data_pivot_s['count']*100
data_pivot_s

count  sum    dolg_s
family_status                              
Не женат / не замужем   2796  273  9.763948
в разводе               1189   84  7.064760
вдовец / вдова           951   63  6.624606
гражданский брак        4134  385  9.313014
женат / замужем        12261  927  7.560558

**Вывод:** По полученным результатам можно сделать вывод, что больше всего задолженностей среди тех клиентов, кто "не женат/не замужем", чуть меньше среди тех, кто состоит в "гражданском браке". Меньше всего задолженностейм у тех клиентов, кто "вдовец/вдова". Кол-во задолженностей у тех клиентов кто, "женат/замужем" и тех кто "в разводе" различаются на 0,5%

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [139]:
dol_d = data.groupby('total_income_category')['debt'].sum() #считаем кол-во должников в каждой категории доходов
kol_d = data.groupby('total_income_category')['gender'].count() #считаем общее кол-во клиентов в каждой категории доходов
print(dol_d)
print(kol_d)
for i in data['total_income_category'].unique():
    convers_d = (dol_d/kol_d)*100   #считаем конверсию 
convers_d

total_income_category
A       2
B     354
C    1353
D      21
E       2
Name: debt, dtype: int64
total_income_category
A       25
B     5014
C    15921
D      349
E       22
Name: gender, dtype: int64


total_income_category
A    8.000000
B    7.060231
C    8.498210
D    6.017192
E    9.090909
dtype: float64

In [140]:
data_grouped_d = data.groupby(['total_income_category']).agg({'gender':'count','debt':'sum'})
data_grouped_d['dolg_d'] = data_grouped_d['debt'] / data_grouped_d['gender'] *100
data_grouped_d

gender  debt    dolg_d
total_income_category                        
A                          25     2  8.000000
B                        5014   354  7.060231
C                       15921  1353  8.498210
D                         349    21  6.017192
E                          22     2  9.090909

In [141]:
data_pivot_d = data.pivot_table(index=['total_income_category'], values=('debt'), aggfunc=('count','sum'))
data_pivot_d['dolg_d']=data_pivot_d['sum']/data_pivot_d['count']*100
data_pivot_d

count   sum    dolg_d
total_income_category                       
A                         25     2  8.000000
B                       5014   354  7.060231
C                      15921  1353  8.498210
D                        349    21  6.017192
E                         22     2  9.090909

**Вывод:** Из полученных данных можно сделать вывод, что кол-во клиентов с зарплатой меньше 30000 (категория E) и больше 1000001
(категория A) очень мало, поэтому выборка будет не показательна. По остальным данным можно сделать вывод, что больше всего задолженностей у клиентов с доходом от 50000 до 200000 (категория C),а меньше всего у клиентов с доходом от 30001 до 50000(категория D).

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

In [142]:
dol_p = data.groupby('purpose_category')['debt'].sum() #считаем кол-во должников в каждой категории целей
kol_p = data.groupby('purpose_category')['gender'].count() #считаем общее кол-во клиентов в каждой категории целей
print(dol_p)
print(kol_p)
for i in data['purpose_category'].unique():
    convers_p = (dol_p/kol_p)*100   #считаем конверсию 
convers_p

purpose_category
операции с автомобилем      400
операции с недвижимостью    780
получение образования       369
проведение свадьбы          183
Name: debt, dtype: int64
purpose_category
операции с автомобилем       4279
операции с недвижимостью    10751
получение образования        3988
проведение свадьбы           2313
Name: gender, dtype: int64


purpose_category
операции с автомобилем      9.347978
операции с недвижимостью    7.255139
получение образования       9.252758
проведение свадьбы          7.911803
dtype: float64

In [150]:
data_grouped_p = data.groupby(['purpose_category']).agg({'gender':'count','debt':'sum'})
data_grouped_p['dolg_p'] = data_grouped_p['debt'] / data_grouped_p['gender'] *100
data_grouped_p

gender  debt    dolg_p
purpose_category                                
операции с автомобилем      4279   400  9.347978
операции с недвижимостью   10751   780  7.255139
получение образования       3988   369  9.252758
проведение свадьбы          2313   183  7.911803

In [147]:
data_pivot_p = data.pivot_table(index=['purpose_category'], values=('debt'), aggfunc=('count','sum'))
data_pivot_p['dolg_p']=data_pivot_p['sum']/data_pivot_p['count']*100
data_pivot_p



count  sum    dolg_p
purpose_category                              
операции с автомобилем     4279  400  9.347978
операции с недвижимостью  10751  780  7.255139
получение образования      3988  369  9.252758
проведение свадьбы         2313  183  7.911803

**Вывод:** Больше задолженностей у тех клиентов, кто брал кредит в категории "операции с автомобилем", а также на "получение образования". Меньше всего задолженностей у клиентов, кто брал кредит на "операции с недвижимостью" и на "проведение свадьбы". 

<div class="alert alert-success">
<h3> Комментарий ревьюера ✔️<a class="tocSkip"> </h3>

С выводом согласен. Как думаешь, почему кредиты на недвижимость и свадьбы в среднем лучше выплачиваются?
</div>

<div class="alert alert-block alert-info">
    
Кредиты на недвижимость и свадьбы в среднем лучше выплачиваются, потому что чаще всего их берет пара(или семья), где как правило два источника дохода, если работают оба супруга.

    
</div>

<div class="alert alert-success">
<h3> Комментарий ревьюера v2 ✔️<a class="tocSkip"> </h3>

 Согласен с тобой. В связи с этим снижаются риски невыплат.
</div>

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

*Ответ:* Данные были пропущены в столбцах days_employed (общий трудовой стаж в днях) и total_income (ежемесячный доход), все это числовые значения. Возможно, причиной пропуска в данных мог быть человеческий фактор, когда человек вводил данные и просто пропустил (умышленно или нет) данные поля для ввода, если они были обозначены как необязательные.  Или могла произойти технологическая ошибка при копировании, считывании, записи или смене формата данных.   

<div class="alert alert-success">
<h3> Комментарий ревьюера ✔️<a class="tocSkip"> </h3>

Абсолютно верно. Основные причины пропусков - это технические проблемы и человеческий фактор.
</div>

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:* Для заполнения пропусков в столбцах days_employed (общий трудовой стаж в днях) и total_income (ежемесячный доход) лучше использовать медиану, чем среднее арифметическое, потому что в таких данных могут быть выдающиеся значения, которые сместят среднее и результат окажется не таким как ожидался. 

<div class="alert alert-success">
<h3> Комментарий ревьюера ✔️<a class="tocSkip"> </h3>

Да, верно, такие сильно выделяющиеся значения ещё называют выбросами.
</div>

### Шаг 4: общий вывод.

Из проведенного анализа и полученных результатов можно сделать вывод, что на факт погашения кредита влияет и семейное положение и количество детей клиента. Больше всего берут кредит клиенты без детей при этом, у них самый низкий процент задолженностей 7,5%. Меньше всего берут кредит клиенты, кто остался "вдовцом/вдовой", и у них также самый низкий процкент задолженностей 6,6% . 
Чаще всего кредит берут клиенты с доходом от 50000 до 200000 (категория C), при этом процент задолженностей среди них 8,5%.
Самая распространенная категория взятия кредита "операции с недвижимостью", при этом процент задолженностей в этой категории клиентов самый маленький 7,3%.



<div class="alert alert-block alert-info">
    
ФИНАЛЬНЫЙ ВЫВОД

    
</div>

Передо мной была поставлена задача от кредитного отдела банка. Требовалось выяснить,**влияет ли семейное положение и количество детей клиента на факт погашения кредита в срок**. Входные данные, полученные от банка — статистика о платёжеспособности клиентов. Результаты исследования будут учтены при построении модели кредитного скоринга — специальной системы, которая оценивает способность потенциального заёмщика вернуть кредит банку.

Решение данной задачи состояло из нескольких этапов. 

На первом этапе была изучена общая информация о данных и выполнена их предобработка. 
Изучив данные, были обнаружены некоторые проблемы, в частности  были обнаружены пропуски в столбцах days_employed (общий трудовой стаж в днях) и total_income (ежемесячный доход), все это числовые значения. Возможно, причиной пропуска в данных мог быть человеческий фактор, или технологическая ошибка. Для заполнения пропусков в этих столбцах использовали медиану, чтобы имеющиеся выбросы не оказали влияния на общую картину.

Также были устранены артефакты в данных об общем трудовом стаже в днях, имелись отрицательные значения. И удалены аномальные значения в столбце о кол-ве детей клиента. Проверены типы данных.

Также в столбце об образовании встречались дубликаты, написанные разным регистром, данные дубликаты были устранены.
Для удобства анализа данные о доходах клиента были разбиты на 4-ре категории:

- 0–30000 — 'E';
- 30001–50000 — 'D';
- 50001–200000 — 'C';
- 200001–1000000 — 'B';
- 1000001 и выше — 'A'.

Также были выделены 4-ре категории в зависимости от целей взятия кредита:

- 'операции с автомобилем',
- 'операции с недвижимостью',
- 'проведение свадьбы',
- 'получение образования'.

Для того чтобы ответить на поставленные в задаче вопросы на этапе анализа данных были рассмотрены несколько зависимостей между данными.

Для рассчета использовали конверсию, отношение числа должников в определенной категории к общему числу клиентов в этой же категории.

Сначала рассмотрели ***зависимость между количеством детей и возвратом кредита в срок**:

Из полученных результатов видно, что больше всего задолженностей у тех клиентов, кто имеет **1-го (9,23%)**, **2 -х(9,45%)** или **4 -х (9,76%)** детей. Клиенты с 5-ю детьми в расчете не учитываем, так как статистика по данной группе слишком мала. **Наименьшее кол-во задолженностей среди клиентов без детей (7,54%)**. Процент должников среди клиентов с **3-мя детьми (8,18%)**.

Также была рассмотрена ***зависимость между семейным положением и возвратом кредита в срок***:

По полученным результатам можно сделать вывод, что больше всего задолженностей среди тех клиентов, кто **"не женат/не замужем" (9,76%)**, чуть меньше среди тех, кто состоит в **"гражданском браке" (9,31%)**. Меньше всего задолженностейм у тех клиентов, кто **"вдовец/вдова" (6,62%)**. Кол-во задолженностей у тех клиентов кто, **"женат/замужем" (7,56%)** и тех кто **"в разводе"(7,06%)** различаются на 0,5%.


Проанализировали ***зависимость между уровнем дохода и возвратом кредита в срок***:

В полученных результатах видно, что кол-во клиентов с зарплатой меньше 30000 (категория E), больше 1000001 (категория A) и с доходом от 30001 до 50000(категория D) очень мало, поэтому выборка будет не показательна и не учитывается в данных расчетах.
По оставшимся данным можно сделать вывод, что больше всего задолженностей у клиентов с доходом **от 50000 до 200000 (категория C) (8,5%)**, а меньше всего у клиентов с доходом **200001–1000000 (категория B)(7.06%)**

В конце определили ***как цели кредита влияют на его возврат в срок***:


Выявили, что больше задолженностей у тех клиентов, кто брал кредит в категории **"операции с автомобилем" (9,35%)**, а также на **"получение образования" (9,25%)**. Меньше всего задолженностей у клиентов, кто брал кредит на **"операции с недвижимостью"(7,26%)** и на **"проведение свадьбы" (7,9%)**.

Из проделанной работы можно сделать вывод, что на факт погашения кредита в срок влияют семейное положение клиента, кол-во детей у клиента, его доход и цели взятия кредита. 
Клиент с минимальными рисками для банка будет выглядеть следующим образом: это клиент без детей, являющийся вдовцом/вдовой, с доходом от 200001–1000000 и взявший кредит на операции с недвижимостью. 
Наиболее рискованный клиент для банка : клиент с 4-мя детьми, который не женат/не замужем, с доходом от 50000 до 200000 и взявший кредит на операции с автомобилем.







Надеюсь в этот раз отправится полная версия моей работы))

<div class="alert alert-danger">
    <h3> Комментарий ревьюера ❌<a class="tocSkip"> </h3>
    
Верно, что затрагиваешь все вопросы исследовательской части. Но описать только лучшего клиента недостаточно.
        
Стоит учитывать, что твой потенциальный руководитель, в первую очередь, обратит внимание на общий вывод. Поэтому важно отразить в нём все основные результаты из всех шагов проекта.
        
Стоит опираться на цели, которые были сформулированы в начале проекта, и ответить на все вопросы. Все ответы стоит подкреплять числами. А также важно описать все промежуточные выводы, которые не описаны в целях, иначе они рискуют остаться незамеченными.
        
Также важно отразить, с какими данными работали, какие проблемы нашли на этапе предобработки и как с ними справились.
        
Рекомендую опираться на такую структуру финального вывода во всех проектах:
* предобработка данных с описанием найденных проблем и путей их решения
* ответы на цели проекта
* промежуточные выводы, которые считаешь важными

Если захочешь поработать на оформлением, советую изучить [краткое руководство по Маркдауну](https://paulradzkov.com/2014/markdown_cheatsheet/).
 
</div>

<div class="alert alert-block alert-danger">
<h3> Комментарий ревьюера v2 ❌<a class="tocSkip"></h3>

Как я писал выше, рассмотрения только хорошего клиента недостаточно. 
    
Важно также показать, как группы по каждому вопросу отличаются друг от друга. Одна группа не задаёт контекста, поэтому цельная картина не может сложиться. И не хватает информации о предобработке.
    
Доработай, пожалуйста, финальный вывод.
</div>

<div class="alert alert-success">
<h3> Комментарий ревьюера v3 ✔️<a class="tocSkip"> </h3>

Теперь получился очень хороший структурированный вывод. Все этапы проекта рассмотрены, а категории клиентов учтены. И даже даны рекомендации для заказчика по хорошему и плохому клиентам. Очень круто, молодец!

</div>